pkg/reconcilers/deploymentconfig.go

```diff
@@ -0,0 +1,89 @@
+package reconcilers
+
+import (
+	"fmt"
+
+	"github.com/3scale/3scale-operator/pkg/common"
+	"github.com/3scale/3scale-operator/pkg/helper"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	appsv1 "github.com/openshift/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+func CreateOnlyDeploymentConfigMutator(existingObj, desiredObj common.KubernetesObject) (bool, error) {
+	return false, nil
+}
+
+func DeploymentConfigResourcesMutator(existingObj, desiredObj common.KubernetesObject) (bool, error) {
+	existing, ok := existingObj.(*appsv1.DeploymentConfig)
+	if !ok {
+		return false, fmt.Errorf("%T is not a *appsv1.DeploymentConfig", existingObj)
+	}
+	desired, ok := desiredObj.(*appsv1.DeploymentConfig)
+	if !ok {
+		return false, fmt.Errorf("%T is not a *appsv1.DeploymentConfig", desiredObj)
+	}
+
+	return DeploymentConfigContainerResourcesReconciler(desired, existing), nil
+}
+
+func GenericDeploymentConfigMutator(existingObj, desiredObj common.KubernetesObject) (bool, error) {
+	existing, ok := existingObj.(*appsv1.DeploymentConfig)
+	if !ok {
+		return false, fmt.Errorf("%T is not a *appsv1.DeploymentConfig", existingObj)
+	}
+	desired, ok := desiredObj.(*appsv1.DeploymentConfig)
+	if !ok {
+		return false, fmt.Errorf("%T is not a *appsv1.DeploymentConfig", desiredObj)
+	}
+
+	update := false
+
+	tmpUpdate := DeploymentConfigReplicasReconciler(desired, existing)
+	update = update || tmpUpdate
+
+	tmpUpdate = DeploymentConfigContainerResourcesReconciler(desired, existing)
+	update = update || tmpUpdate
+
+	return update, nil
+}
+
+func DeploymentConfigReplicasReconciler(desired, existing *appsv1.DeploymentConfig) bool {
+	update := false
+
+	if desired.Spec.Replicas != existing.Spec.Replicas {
+		existing.Spec.Replicas = desired.Spec.Replicas
+		update = true
+	}
+
+	return update
+}
+
+func DeploymentConfigContainerResourcesReconciler(desired, existing *appsv1.DeploymentConfig) bool {
```

Comment:
What's the difference between a "Reconciler" and a "Mutator" in this case? why haven't we called this "DeploymentConfigContainerResourcesMutator"?

Reply:
Mutators functions must belong to the type `MutateFn`

```
type MutateFn func(existing, desired common.KubernetesObject) (bool, error)
```

Reply:
I understand that, I was just wondering what criteria do you use to decide whether you want to make it a MutateFn directly or not. Why not making it a MutateFn function?

Reply:
Could have the same signature. But then you would have to type assert from `common.KubernetesObject` to `*appsv1.DeploymentConfig`.  Maybe they should be private functions of the package?

Reply:
I guess it does not really matter that much. If they are not going to be used outside the package there's no need to export them. Up to you I'm fine with both approaches.

Reply:
Making them private, then.

Reply:
They cannot be private. Some of them, for instance `DeploymentConfigResourcesMutator`, are being used by mutators not included in this package.